src/top-issues/top-issues.service.ts

```diff
@@ -0,0 +1,113 @@
+import { BadRequestException, Injectable, InternalServerErrorException, Logger, NotFoundException } from '@nestjs/common';
+import { CreateTopIssueDto, TopIssueOutputDto, UpdateTopIssueDto } from './schemas/topIssues.schema';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { TopIssue } from '@prisma/client';
+import { PrismaClientValidationError } from '@prisma/client/runtime/library';
+
+@Injectable()
+export class TopIssuesService {
+  private readonly logger = new Logger(TopIssuesService.name);
+  constructor(private prismaService: PrismaService) {}
+
+  async create(body: CreateTopIssueDto): Promise<TopIssueOutputDto> {
+    const { name, icon } = body;
+
+    try {
+      const { id } = await this.prismaService.topIssue.create({
+        data: {
+          name, 
+          icon
+        }
+      })
+
+      return {
+        id,
+        name,
+        icon
+      }
+    } catch (error: unknown) {
+      if (error instanceof PrismaClientValidationError) {
+        this.logger.error(`Validation error ${error.message}`, error.stack);
+        throw new BadRequestException('Failed to create Top Issue - ' + error.message);
+      }
+      if (error instanceof Error) {
+        this.logger.error(`Failed to create Top Issue ${error.message}`, error.stack);
+        throw new InternalServerErrorException('An unexpected error occured.');
+      }
+      throw error;
+    }
+  }
```

Comment:
I say go ahead and implement that root application prisma filter, for the plethora of reasons you and I discussed and this way we don't have all these `try-catches` all over the place.